addons/mail/models/mail_thread.py

```diff
@@ -1143,7 +1158,7 @@ def message_route(self, message, message_dict, model=None, thread_id=None, custo
         if rcpt_tos_localparts:
             # no route found for a matching reference (or reply), so parent is invalid
             message_dict.pop('parent_id', None)
-            dest_aliases = Alias.search([('alias_name', 'in', rcpt_tos_localparts)])
+            dest_aliases = Alias.search(alias_domain)
```

Comment:
Isn't it simply the same domain as before ? Normally if you are here, it is not a reply or there is no alias of the same model as the parent message (otherwise it should have been considered as a reply). Am I wrong ?

Reply:
No, you're right. Let me change this.